pyrit/prompt_converter/zalgo_converter.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import random
+from typing import Optional
+
+from pyrit.models import PromptDataType
+from pyrit.prompt_converter import ConverterResult, PromptConverter
+
+# Unicode combining characters for Zalgo effect (U+0300–U+036F)
+ZALGO_MARKS = [chr(code) for code in range(0x0300, 0x036F + 1)]
+# Setting a max intensity so people don't do anything insane
+MAX_INTENSITY = 100
+logger = logging.getLogger(__name__)
+
+
+class ZalgoConverter(PromptConverter):
+    def __init__(self, *, intensity: int = 10, seed: Optional[int] = None) -> None:
+        """
+        Initializes the Zalgo converter.
+
+        Args:
+            intensity (int): Number of combining marks per character (higher = more cursed). Default is 10.
+            seed (Optional[int]): Optional seed for reproducible output.
+        """
+        self._intensity = intensity
+        self._seed = seed
+
+    def _normalize_intensity(self) -> int:
+        if self._intensity > MAX_INTENSITY:
+            logger.warning("This is higher intensity than is supported (max 100)")
```

Comment:
```suggestion
            logger.warning(f"ZalgoConverter supports intensity between 0 and 100, but received a value of {self._intensity}. Normalizing to {normalized_intensity}")
```

I would like to add a bit more info to the warning.

At the same time, should we do the normalization in the constructor? It feels odd to do the min-max and warning printing over and over again for every `convert_async` call

Reply:
We also need to make sure it's actually an int by casting to `int(...)`. If it's a float the `randint` call will fail

Reply:
Added normalizing to the constructor, cleaned up logging, and ensured that it is an int, added throwing an error if it cannot be cast to an int.

Reply:
Added the testing for non-int cases